src/ConcordiumNetSdk/SignKey/Ed25519SignKey.cs

```diff
@@ -3,64 +3,91 @@
 namespace ConcordiumNetSdk.SignKey;
 
 /// <summary>
-/// Represents a hex encoded ed25519 sign key.
+/// An ed25519 sign key.
 /// </summary>
 public record Ed25519SignKey : ISigner
 {
     private const int StringLength = 64;
     private const int BytesLength = 32;
 
+    /// <summary>
+    /// Representation of the ed25519 sign key as a length-32 hex string.
+    /// </summary>
     private readonly string _formatted;
+
```

Comment:
Why not simply hold it as the `Key`? And then check that you can construct this when creating it from either bytes or the hex encoded string. This way the user is informed (before signing) that the key is invalid.